src/openms/include/OpenMS/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.h

```diff
@@ -207,6 +207,34 @@ namespace OpenMS
       std::vector<MSSpectrum>& annotated_spectra
     ) const;
 
+    /**
+      @brief Annotates the MS2 spectra with the likely MS1 feature that it was derived from
+
+      Add support for annotating based off of MS1 features resulting from SearchAccurateMass.
+      In this case, the input will be e.g., const FeatureMap& ms1_features and the RTs and names (i.e., PeptideRef),
+      defined in the FeatureMap.
+
+      @param[in] spectra The spectra to filter
+      @param[in] ms1_features the MS1 features
+      @param[out] ms1_features the MS2 features
+      @param[out] annotated_spectra the resulting annotated spectra
+    */
+    void annotateSpectra(
+        const std::vector<MSSpectrum>& spectra,
+        const FeatureMap& ms1_features,
+        FeatureMap& ms2_features,
+        std::vector<MSSpectrum>& annotated_spectra) const;
+    
+    /**
+      @brief Search accurate masses and build add the the peptide hits as features/sub-features
+
+      @param[in] feat_map The feature map to search in
+      @param[in] feat_map_output The output feature map, with peptide identifaction as sub features
+    */
+    void searchSpectrum(
```

Comment:
I think I have not yet understood what this is used for - based on the description.